### PR TITLE
[NBS Docs] Refactor Top Level Permissions Docs

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

This PR attempt to refactor the top level permissions documentation to work with the new backend system.

As @Parsifal-M had also been working on the [same area](https://github.com/backstage/backstage/pull/25799) we'll try and consolidate our work on this.

The intention is to have a follow up PR for the [Plugin Auth Tutorial](https://backstage.io/docs/permissions/plugin-authors/01-setup).

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/permissions/custom-rules.md [hunk 1]
```diff
@@ -135,25 +139,49 @@ class TestPermissionPolicy implements PermissionPolicy {
...
-  env: PluginEnvironment,
-): Promise<Router> {
-  const builder = await CatalogBuilder.create(env);
-  /* highlight-add-next-line */
-  builder.addPermissionRules(isInSystemRule);
-  ...
-  return router;
-}
-```
+The api for providing custom rules may differ between plugins, but there should typically be some extension point that you can use to create a module that adds your rule. For the catalog, this extension point is exposed via `catalogPermissionExtensionPoint`. Here's the steps you'll need to take to add the `isInSystemRule` we created above to the catalog:
```
Comment (Parsifal-M) [inline docs/permissions/custom-rules.md:142]: Could be nice here to link to https://backstage.io/docs/backend-system/architecture/extension-points/ 

Maybe a NIT? `but there should typically be an extension point that you can [use/consume?] in your created module to add your rule`

Its also fine the way it is, maybe it just sounds better in my head 😅

File: docs/permissions/getting-started.md [hunk 2]
```diff
@@ -38,82 +26,75 @@ Like many other parts of Backstage, the permissions framework relies on informat
...
+2. Now we need to add the `@backstage/backend-plugin-api` package:
 
-   ```bash title="From your Backstage root directory"
-   yarn --cwd packages/backend add @backstage/plugin-permission-backend
+   ```bash title="from your Backstage root directory"
+   yarn --cwd packages/backend add @backstage/backend-plugin-api
    ```
 
-2. Add the following to a new file, `packages/backend/src/plugins/permission.ts`. This adds the permission-backend router, and configures it with a policy which allows everything.
+3. Next we will create an `extensions` folder under `packages/backend/src`
```
Comment (Parsifal-M) [inline docs/permissions/getting-started.md:51]: Just wondering, will this be the recommended setup in terms of file structure? although in the end its the integrators choice, I think at least having a "recommended" way of doing it is good! 👍

File: docs/permissions/getting-started.md [hunk 3]
```diff
@@ -38,82 +26,75 @@ Like many other parts of Backstage, the permissions framework relies on informat
 
 [The IdentityResolver docs](../auth/identity-resolver.md) describe the process for resolving group membership on sign in.
 
-## Optionally add cookie-based authentication
-
-Asset requests initiated by the browser will not include a token in the `Authorization` header. If these requests check authorization through the permission framework, as done in plugins like TechDocs, then you'll need to set up cookie-based authentication. Refer to the ["Authenticate API requests"](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md) tutorial for a demonstration on how to implement this behavior.
+## Test Permission Policy
 
-## Integrating the permission framework with your Backstage instance
+To get help validate the permission framework is setup we'll create a Test Permission Policy:
```
Comment (Parsifal-M) [inline docs/permissions/getting-started.md:31]: Think this should be: `To help validate`

File: docs/permissions/getting-started.md (hunk 2 above)
Comment (awanlin) [reply docs/permissions/getting-started.md:51]: Honestly in a perfect world this should be in its own module, we don't have a `yarn new` template for that yet so this was the simplest way to move forward. Based on the need from the Community for these docs to be updated sooner than later I felt like this was the best middle ground.

File: docs/permissions/custom-rules.md (hunk 1 above)
Comment (awanlin) [reply docs/permissions/custom-rules.md]: Added a link and adjusted to match what you suggested. Can you give it another look to make sure I got it right, please?

File: docs/permissions/getting-started.md (hunk 3 above)
Comment (awanlin) [reply docs/permissions/getting-started.md]: Good catch, fixed it 👍

**Review by Parsifal-M — APPROVED** [review]
Nice! I think this will be really useful 🚀  kudos!

File: docs/permissions/getting-started.md (hunk 2 above)
Comment (Parsifal-M) [reply docs/permissions/getting-started.md:51]: Clear! was just checking, I'm also fine with this :)

**Comment by awanlin** [comment]
Hi @Rugvip, sorry for the ping, I know there's a backlog of PRs to review but this would be really helpful for the Community to have in place.

**Review by Rugvip — APPROVED** [review]
Sorry, completely dropped off my radar!

Thank you and let's :shipit: 🎉

**Pull request merged**